pkg/storage/storage.go

```diff
@@ -325,17 +354,31 @@ func (s *storageModule) CreateFilesystem(name string, size uint64, poolType pkg.
 func (s *storageModule) ReleaseFilesystem(name string) error {
 	log.Info().Msgf("Deleting volume %v", name)
 
-	for idx := range s.volumes {
-		filesystems, err := s.volumes[idx].Volumes()
+	for _, pool := range s.pools {
+		if _, mounted := pool.Mounted(); !mounted {
+			continue
+		}
+
+		filesystems, err := pool.Volumes()
 		if err != nil {
 			return err
 		}
-		for jdx := range filesystems {
-			if filesystems[jdx].Name() == name {
-				log.Debug().Msgf("Removing filesystem %v in volume %v", filesystems[jdx].Name(), s.volumes[idx].Name())
-				return s.volumes[idx].RemoveVolume(filesystems[jdx].Name())
+		for _, fs := range filesystems {
+			if fs.Name() == name {
+				log.Debug().Msgf("Removing filesystem %v in volume %v", fs.Name(), pool.Name())
+				err = pool.RemoveVolume(fs.Name())
+				if err != nil {
+					return err
+				}
 			}
 		}
+
+		// After releasing a filesystem by name, there is a chance there is going to be
+		// no reserved space on a volume, in this case we want to shutdown the disk in order to save power
+		if err := s.shutdownUnusedPools(); err != nil {
```

Comment:
do you have to call `shutdownUnusedPools` which loop over everything here ? Can't you just check if `filesystems` had a length of 1 and if this is the case, then we know there are no more subvolume and we can just call `pool.Shutdown` directly ?